infra/build/functions/build_project.py

```diff
@@ -224,9 +224,13 @@ def get_build_steps(project_name, project_yaml_file, dockerfile_lines,
                     # `cd /src && cd {workdir}` (where {workdir} is parsed from
                     # the Dockerfile). Container Builder overrides our workdir
                     # so we need to add this step to set it back.
-                    ('rm -r /out && cd /src && cd {workdir} && mkdir -p {out} '
-                     '&& compile || (echo "{failure_msg}" && false)'
-                    ).format(workdir=workdir, out=out, failure_msg=failure_msg),
+                    # Reset HOME so that it doesn't point to a persisted volume
+                    # (see https://github.com/google/oss-fuzz/issues/6035).
+                    ('export HOME=/root; rm -r /out && cd /src && cd {workdir} '
```

Comment:
is /root even writeable for current user, maybe some folder in /tmp ?

Reply:
It is. /root is actually the default. it gets overridden by gcb so I'm setting it back to the original value.

Reply:
Set HOME in `env` above instead?

Reply:
HOME is already set by GCB. By treating home as an env var, we're relying on 1. GCB setting our value second on the commandline. 2. docker overwriting the first value of HOME with the second one. I think it makes more sense to set it here since we know how bash will behave but I'm open to changing this if you still think I made the wrong choice.

Reply:
HOME is already set by GCB. By treating home as an env var, we're relying on 1. GCB setting our value second on the commandline. 2. docker overwriting the first value of HOME with the second one. I think it makes more sense to set it here since we know how bash will behave but I'm open to changing this if you still think I made the wrong choice.

Reply:
I think it's much cleaner to keep in `env`. These are reasonable assumptions to make, and makes all our steps more consistent. Otherwise it may come back in the future if some other step is broken because of some persistence in home.